pkgs/development/compilers/chicken/5/overrides.nix

```diff
@@ -166,6 +165,46 @@ in
   pledge = addMetaAttrs { platforms = lib.platforms.openbsd; };
   unveil = addMetaAttrs { platforms = lib.platforms.openbsd; };
 
+  # overrides for chicken 5.4
+  dbus = pkg: old:
+    (addToBuildInputsWithPkgConfig [ pkgs.dbus ] old) // {
+      # backticks in compiler options
+      # aren't supported anymore as of chicken 5.4, it seems.
+      preBuild = ''
+        pkg-config --cflags dbus-1 > dbus_cflags
+
+        pkg-config --libs dbus-1 > dbus_libs
+
+        script='{getline cflags < "dbus_cflags";
+                 getline libs < "dbus_libs";
+                 gsub(/`pkg-config --cflags dbus-1`/, cflags);
+                 gsub(/`pkg-config --libs dbus-1`/, libs);
+                }1'
+
+        awk "$script" dbus.egg > temp.egg
+        mv temp.egg dbus.egg
+        awk "$script" dbus.setup > temp.setup
+        mv temp.setup dbus.setup
+
+        rm dbus_cflags
+        rm dbus_libs
+      '';
+    };
+  math = old: pkg: {
+    # define-values is used but not imported
+    # some breaking change happened now it needs to be done
+    # explicitly?
+    preBuild = ''
+      find . ! \( -path ./.git -prune \) -type f | xargs sed -i 's/only chicken.base/only chicken.base define-values/'
```

Comment:
Could you try with something like: 
```
substituteInPlace *.scm **/*.scm \
  --replace-fail 'only chicken.base' 'only chicken.base define-values'
```
?

If it does not work for some reason it's not a problem.

Reply:
Speak of the devil, I found out about `substituteInPlace` at the same time you sent this. `--replace-quiet` I think is the way to go, pushing

Reply:
Whoops, it didn't go through the first time, I don't know why :/, it's in there now